src/phpDocumentor/Reflection/Php/Factory/File/CacheMiddleware.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Reflection\Php\Factory\File;
+
+use phpDocumentor\Reflection\Middleware\Middleware;
+use phpDocumentor\Reflection\Php\File;
+use Stash\Item;
+use Stash\Pool;
+
+final class CacheMiddleware implements Middleware
+{
+    /**
+     * Cache namespace used for this repository.
+     */
+    const CACHE_NAMESPACE = 'Documentation\\Api\\Php';
```

Comment:
How will this work with multiple versions? Will versions mix their caches? (perhaps something to address in a next PR)